javalib/src/main/scala/java/util/concurrent/ConcurrentHashMap.scala

```diff
@@ -0,0 +1,205 @@
+package java.util.concurrent
+
+import scala.collection.JavaConversions._
+
+trait ConcurrentMap[K, V] extends java.util.Map[K, V] {
+
+  def remove(key: Any, value: Any): Boolean = {
+    if (containsKey(key) && get(key).equals(value)) {
+      remove(key)
+      true
+    } else false
+  }
+
+  def replace(key: K, value: V): V = {
+    if (containsKey(key)) {
+      put(key, value)
+    } else get(key)
+  }
+
+  def replace(key: K, oldValue: V, newValue: V): Boolean = {
+    if (containsKey(key) && get(key).equals(oldValue)) {
+      put(key, newValue)
+      true
+    } else false
+  }
+}
+
+class ConcurrentHashMap[K >: Null, V >: Null](initialCapacity: Int,
+    loadFactor: Float,
+    concurrencyLevel: Int,
+    initialMap: java.util.Map[_ <: K, _ <: V])
+    extends java.util.AbstractMap[K, V]
+    with java.util.concurrent.ConcurrentMap[K, V]
+    with java.io.Serializable {
+
+  def this() = 
+    this(16, 0.75F, 16, mapAsJavaMap(Map[K,V]()))
+
+  def this(initialCapacity: Int) = 
+    this(initialCapacity, 0.75F, 16, mapAsJavaMap(Map[K,V]()))
+
+  def this(initialCapacity: Int, loadFactor: Float) = 
+    this(initialCapacity, loadFactor, 16, mapAsJavaMap(Map[K,V]()))
+
+  def this(initialCapacity: Int, loadFactor: Float, concurrencyLevel: Int) = 
+    this(initialCapacity, loadFactor, concurrencyLevel, mapAsJavaMap(Map[K,V]()))
```

Comment:
Make this constructor be the primary constructor (in the signature of the class). The constructor taking an `initialMap` can first call `this()` then call `putAll(initialMap)`. This avoids all the `mapAsJavaMap(Map[K, V]())`.